config/initializers/paperclip.rb

```diff
@@ -0,0 +1,19 @@
+if Rails.env.production?
+  Paperclip::Attachment.default_options.update({
+    storage: :s3,
+    s3_permissions: :private,
+    s3_credentials: {
+      bucket: ENV.fetch('S3_BUCKET_NAME'),
+      access_key_id: ENV.fetch('AWS_ACCESS_KEY_ID'),
+      secret_access_key: ENV.fetch('AWS_SECRET_KEY_ID'),
+      s3_region: ENV.fetch('AWS_REGION')
+    },
+    s3_host_name: ENV.fetch('S3_HOST_NAME'),
+    url: ':s3_domain_url',
+    path: '/:class/:attachment/:id_partition/:filename'
+  })
+else
+  Paperclip::Attachment.default_options[:path] = ":rails_root/public/upload_images/:id_partition/:filename"
```

Comment:
development の時は設定を上書きしているということですか?

Reply:
通信量節約のため、dev環境の時はS3と連携せずローカルに保存する、という設定にしています。

Reply:
S3を利用するにあたりコントローラ経由の方法に変えた影響か、元々のデフォルト設定のままでは動かなくなっていたので、上書きしました。

Reply:
https://github.com/thoughtbot/paperclip#understanding-storage
```
This location was chosen because, on standard Capistrano deployments, the public/system directory can be symlinked to the app's shared directory, meaning it survives between deployments. For example, using that :path, you may have a file at
```

Rails 界隈では public/system を使うケースが多いので、 upload_images => system でお願いします。(Carrierwave, Dragonfly も public/system です)